app/models/solr_document.rb

```diff
@@ -149,6 +149,17 @@ def uri
     fetch("uri_tesim", [])
   end
 
+  def doi_url
+    uri.each do |link|
+      return link if link.downcase.start_with?('https://doi.org/')
+    end
+    nil
+  end
```

Comment:
I am using the *first* URL that starts with `doi.org` as the "DOI URL". We can tweak this if a separate field should be used or created to handle DOIs.